01_Data/src/layers/sequelize/repository/MessageInfo.ts

```diff
@@ -30,14 +30,26 @@ export class SequelizeMessageInfoRepository extends SequelizeRepository<MessageI
   }
 
   async createOrUpdateByMessageInfoTypeAndStationId(message: MessageInfoType, stationId: string, componentId?: number): Promise<MessageInfo> {
-    const [savedMessageInfo, _messageInfoCreated] = await this.upsert(
-      MessageInfo.build({
+    return await this.s.transaction(async (transaction) => {
+      const messageInfo = MessageInfo.build({
         stationId: stationId,
         componentId: componentId,
         ...message,
         active: true,
-      }),
-    );
-    return savedMessageInfo;
+      });
+      // TODO is this searching correctly?
```

Comment:
As describing in the OCPP 2.0.1, the id of MessageInfoType is unique within an exchange context. It is defined within the OCPP context as a positive Integer value (greater or equal to zero). My understanding is that the id is unique for each charger. So stationId and message.id should be used for finding a unique one. (An composite unique constraint on these 2 field is also set in the messageInfo table.)

Reply:
Got it, changed to search by message.id